### PR TITLE
HB-5933: use os_log instead of print

### PR DESCRIPTION
**Pull request by bwised**
https://chartboost.atlassian.net/browse/HB-5933

### CODE REVIEW

File: Source/ReferenceAdapterConfiguration.swift [hunk 1]
```diff
@@ -4,18 +4,23 @@
...
 
 import Foundation
+import os.log
 
 /// INTERNAL. FOR DEMO AND TESTING PURPOSES ONLY. DO NOT USE DIRECTLY.
 ///
 /// A list of externally configurable properties pertaining to the partner SDK that can be retrieved and set by publishers.
 @objc public class ReferenceAdapterConfiguration: NSObject {
     
+    private static var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Configuration")
```
Comment (bwised) [inline Source/ReferenceAdapterConfiguration.swift:14]:
```suggestion
    private static let log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Configuration")
```

File: Source/ReferenceAdapterConfiguration.swift [hunk 2]
```diff
@@ -4,18 +4,23 @@
...
+
     /// Flag that can optionally be set to enable the partner's test mode.
     /// Disabled by default.
     @objc public static var testMode: Bool = false {
         didSet {
             ReferenceSdk.setTestMode(enabled: testMode)
-            print("Reference SDK test mode set to \(testMode)")
+            if #available(iOS 12.0, *) {
+                os_log(.debug, log: log, "Reference SDK test mode set to %{public}s", testMode ? "true" : "false")
```
Comment (bwised) [inline Source/ReferenceAdapterConfiguration.swift:22]:
```suggestion
                os_log(.debug, log: log, "Reference SDK test mode set to %{public}s", "\(testMode)")
```

File: Source/ReferenceAdapterConfiguration.swift [hunk 3]
```diff
@@ -24,7 +29,9 @@ import Foundation
     @objc public static var verboseLogging: Bool = false {
         didSet {
             ReferenceSdk.setVerboseLogging(enabled: verboseLogging)
-            print("Reference SDK verbose logging set to \(verboseLogging)")
+            if #available(iOS 12.0, *) {
+                os_log(.debug, log: log, "Reference SDK verbose logging set to %{public}s", verboseLogging ? "true" : "false")
```
Comment (bwised) [inline Source/ReferenceAdapterConfiguration.swift:33]:
```suggestion
                os_log(.debug, log: log, "Reference SDK verbose logging set to %{public}s", "\(verboseLogging)")
```

File: Source/ReferenceSdk/ReferenceBannerAd.swift [hunk 4]
```diff
@@ -28,7 +29,10 @@ class ReferenceBannerAd: UIView {
     
     /// The click through URL to navigate to after a click event has been processed
     let clickThroughUrl = "https://www.chartboost.com/helium/"
-    
+
+    /// The log configuration.
+    private var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Banner")
```
Comment (bwised) [inline Source/ReferenceSdk/ReferenceBannerAd.swift:34]:
```suggestion
    private lazy var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Banner")
```

File: Source/ReferenceSdk/ReferenceFullscreenAd.swift [hunk 5]
```diff
@@ -27,6 +28,9 @@ class ReferenceFullscreenAd {
     /// The delegate for this Reference ad.
     weak var delegate: ReferenceFullscreenAdDelegate?
         
+    /// The log configuration.
+    private var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Banner")
```
Comment (bwised) [inline Source/ReferenceSdk/ReferenceFullscreenAd.swift:32]:
```suggestion
    private lazy var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Banner")
```

File: Source/ReferenceSdk/ReferenceFullscreenAd.swift [hunk 6]
```diff
@@ -27,6 +28,9 @@ class ReferenceFullscreenAd {
     /// The delegate for this Reference ad.
     weak var delegate: ReferenceFullscreenAdDelegate?
         
+    /// The log configuration.
+    private lazy var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Banner")
```
Comment (daniel-barros) [inline Source/ReferenceSdk/ReferenceFullscreenAd.swift:32]:
```suggestion
    private lazy var log = OSLog(subsystem: "com.chartboost.mediation.adapter.reference", category: "Fullscreen")
```

**Review by daniel-barros — DISMISSED** [review]

File: Source/ReferenceSdk/ReferenceFullscreenAd.swift (hunk 6 above)
Comment (bwised) [reply Source/ReferenceSdk/ReferenceFullscreenAd.swift:32]: Oof, thanks.

**Review by daniel-barros — APPROVED** [review]

**Pull request merged**